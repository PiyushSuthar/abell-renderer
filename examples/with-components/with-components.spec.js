const path = require('path');
const {
  buildAndGetSelector
} = require('../../tests/utils/baseTestFramework.js');

// all the contents are managed by a jest snapshot internally
const TEST_MAP = [
  {
    desc: 'should render content of Nav Component',
    query: '[data-test="nav-component"]'
  },
  {
    desc: 'should render content of Footer Component',
    query: '[data-test="footer-component"]'
  },
  {
    desc: 'should render main content',
    query: '[data-test="main-content"]'
  }
];

describe('examples/with-components', () => {
  let $;
  beforeAll(async () => {
    $ = await buildAndGetSelector({
      exampleToRun: 'with-components',
      outPath: path.join(__dirname, 'out.html')
    });
  });

  for (const test of TEST_MAP) {
    it(test.desc, () => {
      expect($(test.query).html()).toMatchSnapshot();
    });
  }
});
