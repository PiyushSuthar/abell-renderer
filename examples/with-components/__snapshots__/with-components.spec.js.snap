// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples/with-components should render content of Footer Component 1`] = `" This is my <b data-abell-cybowg>Footer</b> nice Test JSON Import Pass"`;

exports[`examples/with-components should render content of Nav Component 1`] = `" This is my Navbar 13 "`;

exports[`examples/with-components should render main content 1`] = `"Content"`;
